pkg/apis/core/v1alpha1/federatedtypeconfig_types.go

```diff
@@ -89,6 +89,35 @@ type FederatedTypeConfig struct {
 	Status FederatedTypeConfigStatus `json:"status,omitempty"`
 }
 
+// FederatedResource is a generic resource representing any federated type
+// +k8s:deepcopy-gen=false
+type FederatedResource struct {
```

Comment:
I'm not sure I understand the need for this generic `FederatedResource` API type. I don't see it being persisted; it looks like it's just used to capture the status into this generic object and then just copied into the specified `Federated<Type>` resource for persistence. Why not work directly with the status for each individual `Federated<Type>` resource that you've added in this PR? Am I missing something?

Reply:
Agree with your argument. Updated accordingly. PTAL. Thanks!